pymatgen/symmetry/analyzer.py

```diff
@@ -1697,3 +1700,543 @@ def __str__(self):
+        bns_number = ".".join([str(x) for x in self._MagneticSpacegroup._data["bns_number"]])
+        return (bns_label, bns_number)  
+
+    def get_og(self):
+        """
+        Get the magnetic spacegroup in the OG notation [1], e.g. (P1, 1.1.1). 
+
+        [1] Opechowski and Guccione (1965). Magnetism, edited by G. T. Rado and H. Suhl, Vol. II, 
+            Part A, pp. 105-165. New York: Academic Press
+      
+        Returns:
+            (str, list(int)): Magnetic space group in a tuple of (symbol, number) for structure.
+        """
+        og_label = self._MagneticSpacegroup._data["og_label"]
+        og_number = ".".join([str(x) for x in self._MagneticSpacegroup._data["og_number"]])
+        return (og_label, og_number)  
+
+    def _same_MagneticStructure(self, struc1, struc2, pos_tolerance=0.0015, magmom_tolerance=0.3):
+        """
+        Compare two pymatgen.core.structure.Structure objects, struc1 and struc2,
+        with respect to their sitesi, species and magnetic moment.
+
+        Takes:
+          struc1           pymatgen.core.structure.Structure 
+          struc1           pymatgen.core.structure.Structure 
+          pos_tolerance    float (default: 0.0015)
+          magmom_tolerance float (default: 0.3)
+
+        Returns:
+          boolean          True if struc1 covers struc2
+          set(tuple(tuple(str), tuple(float, float, float), tuple(float, float, float))) 
+                         Residue: Set of species, fractional coordinates 
+                         and magnetic moments that are distinct. 
+                         Empty set if boolean is True.
+        """
+        same = False
+        nsite = len(struc1.sites)
+        if nsite != len(struc2.sites):
+            res = "number of sites differ"
+            pass
+        else:
+            sites1_prim = [np.mod(i.frac_coords + 10 + 1e-9, 1) - 1e-9 for i in struc1._sites]
+            sites1_prim = [tuple(k) for k in sites1_prim]
+            sites2_prim = [np.mod(i.frac_coords + 10 + 1e-9, 1) - 1e-9 for i in struc2._sites]
+            sites2_prim = [tuple(k) for k in sites2_prim]
+        
+            magms1_cart = [i.properties["magmom"].moment for i in struc1._sites]
+            magms1_cart = [tuple(k.round(3)) for k in magms1_cart]
+            magms2_cart = [i.properties["magmom"].moment for i in struc2._sites]
+            magms2_cart = [tuple(k.round(3)) for k in magms2_cart]
+            species1 = [i._species for i in struc1._sites]
+            species2 = [i._species for i in struc2._sites]
+      
+            set1 = set(tuple([tuple([species1[i],sites1_prim[i],magms1_cart[i]]) for i in range(nsite)]))
+            set2 = set(tuple([tuple([species2[i],sites2_prim[i],magms2_cart[i]]) for i in range(nsite)]))
+            res = self._difference(set1, set2, pos_tolerance=pos_tolerance, magmom_tolerance=magmom_tolerance)
+            if res == set():
+                same = True
```

Comment:
An empty set evaluates as `False` in Python, so this statement could be `if not res` -- this is no more or less correct, but this latter notation is more commonly used.  The variable `res` could be called `diff` to make its purpose more clear.